Add documentation on interactions between multiple views in BIND and the dns_rfc2136 plugin

warewolf opened the pull request:
Follow up to https://github.com/certbot/certbot/pull/9106 :

This is a documentation only pull request, describing how multiple views in BIND can cause certbot to be unable to find the authorative nameserver for a zone, with an example configuration that causes BIND to respond correctly with the AA (authorative answer) flag set for the dns_rfc2136 plugin.

alexzorin reviewed (changes requested):
Thank you very much, I am sure this will be of great benefit to users.

To fix the failing build, the line length needs to be under 100 characters and not have any trailing whitespace. You can break the lines in the source code without affecting how the documentation is rendered, so that should be an easy fix.

The other comment/question I have is whether we want to split this out to a separate page. As it stands, this PR makes the documentation page quite lengthy, and pushes down the examples quite far. We could benefit from splitting it off onto a separate page and adding it to the doctree, like how the `api` page is right now. We could then also reference it from "Sample Bind Configuration" section.

I'm not going to strongly insist we do that, but I'd be keen to hear your thoughts on it.

warewolf commented:
Ok, hopefully this passes checks and addresses (most) of your concerns.  On the *verbosity* of the documentation I added, I wanted to make sure it was clear and rewrote it a bunch to try to make it as terse as possible but if you want to refer to it from this page, and actually put the documentation elsewhere I have no issue with that.

Just as long as the `Unable to determine base domain for _acme-challenge.example.com` message is in there to help direct users to diagnosing what is going wrong.

When I initially set up certbot+rfc2136 many months ago, I was troubleshooting the DNS side of the workflow using `nsupdate` to emulate certbot's adding/removing records ... I could make it longer if you like! :rofl:

alexzorin commented on certbot-dns-rfc2136/certbot_dns_rfc2136/__init__.py:
```diff
@@ -108,35 +140,59 @@
-     -d www.example.com
+.. note::
+   Order matters in BIND views, the ``in-view`` zone option must refer to a
+view defined preceeding it, it cannot refer to a view defined later in the configuration file.
```
```suggestion
   view defined preceeding it, it cannot refer to a view defined later in the configuration file.
```

alexzorin commented on certbot-dns-rfc2136/certbot_dns_rfc2136/__init__.py:
```diff
@@ -108,35 +140,59 @@
 
-.. code-block:: bash
-   :caption: To acquire a certificate for ``example.com``
+If your BIND configuration leverages multiple views, certbot may fail with an
```
```suggestion
If your BIND configuration leverages multiple views, Certbot may fail with an
```

alexzorin commented on certbot-dns-rfc2136/certbot_dns_rfc2136/__init__.py:
```diff
@@ -108,35 +140,59 @@
-.. code-block:: bash
-   :caption: To acquire a single certificate for both ``example.com`` and
-             ``www.example.com``
+One solution is to logically place the zone into the view certbot is sending
```
```suggestion
One solution is to logically place the zone into the view Certbot is sending
```

alexzorin commented on certbot-dns-rfc2136/certbot_dns_rfc2136/__init__.py:
```diff
@@ -108,35 +140,59 @@
-   :caption: To acquire a certificate for ``example.com``
+If your BIND configuration leverages multiple views, certbot may fail with an
+``Unable to determine base domain for _acme-challenge.example.com`` error.
+This error occurs when certbot isn't able to communicate with an authorative
```
```suggestion
This error occurs when Certbot isn't able to communicate with an authorative
```

alexzorin commented on certbot-dns-rfc2136/certbot_dns_rfc2136/__init__.py:
```diff
@@ -108,35 +140,59 @@
+can cause this error.  If the zone is only present in the external view, and
+the credentials_ ``dns_rfc2136_server`` setting is set (e.g. 127.0.0.1) so the
+DNS server's ``match-clients`` view option causes the DNS server to route
+certbot's query to the internal view; the internal view doesn't contain the
```
```suggestion
Certbot's query to the internal view; the internal view doesn't contain the
```

alexzorin reviewed (changes requested):
OK. I am happy with the way you have reorganized the content, we can hold off from creating a separate page.

Just one fix to indentation to fix the build, and some capitalizing Certbot when not referring to it as a command (`certbot`).

warewolf commented:
Suggestions accepted, thank you!

alexzorin reviewed (approved):
Merging from my phone, hopefully this works haha. 

LGTM thanks.
